### ISSUE TITLE
Crash in ClassHelperMethods.swift

### ISSUE THREAD

**Issue by naudecruywagen** [net-a-porter-mobile/XCTest-Gherkin#74]
Hey guys, 

I am experiencing a crash when trying to run tests with the GoogleTagManager cocoapod added to my project.

I've attached a sample project that demonstrates the issue.

Steps to reproduce: (the easy way)

1. Open the attached project's workspace - GherkinCrash.xcworkspace
2. Run the tests: ⌘+U
3. Observe the crash

Steps to reproduce: (the harder way)

1. Create an Xcode project.
2. Add  `pod 'GoogleTagManager', '~> 6.0'` to the Podfile of the project.
3. pod install and open the workspace.
3. Add the XCTest-Gherkin files manually into the project.
4. Write a test using NativeRunner.
5. Try to run the test. 
6. EXC_BAD_ACCESS in ClassHelperMethods

[GherkinCrash.zip](https://github.com/net-a-porter-mobile/XCTest-Gherkin/files/1188671/GherkinCrash.zip)

**Comment by kerrmarin** [comment]
This looks a lot like https://github.com/AFNetworking/AFNetworking/pull/2294 🤔

**Comment by kerrmarin** [comment]
I'm still not sure what causes this, but as a work-around, you can get around this by not linking the `GoogleTagManager` to the test target, e.g.:

```
platform :ios, '10.3'

target 'GherkinCrash' do
  pod 'GoogleTagManager', '~> 6.0'

end

target 'GherkinCrashTests' do
    
end
```

**Comment by nap-sam-dean** [comment]
We've run into this before, and thought we'd fixed it - https://github.com/net-a-porter-mobile/XCTest-Gherkin/issues/56 

Looks like we were wrong :)

The other fix is to move to xcode9 (oddly, it doesn't crash then) but I guess that's not a great fix for you :|

**Comment by naudecruywagen** [comment]
Sorry for the long silence.

@kerrmarin unfortunately this is how the Podfile looks already and it still crashes.

@nap-sam-dean Xcode 9 is also not an option for us yet, but if push comes to shove we'll probably have to do this...

Any other ideas?

**Comment by zacoid55** [comment]
@naudecruywagen I had this problem as well as I ran `gem update` by mistake and I quit out of it prematurely. Have you tried switching to another version of Ruby to test this?

**Comment by deanWombourne** [comment]
Hi, I'm currently looking into a similar bug I get in xcode9 iOS 11, I'll keep you updated how I get on.

**Comment by deanWombourne** [comment]
Hi @naudecruywagen - do you still have this issue. We've got a potential fix, but the release it's fixed on requires xcode9 :( Can you try checking out commit 5aa806d and see if that works for you in Xcode 8?

**Comment by zacoid55** [comment]
@deanWombourne any ideas of when the next release might be?

**Comment by deanWombourne** [comment]
Right now - I've just released 0.11.0 :)

**Comment by naudecruywagen** [comment]
Hi @deanWombourne, I have checked out [5aa806d](https://github.com/net-a-porter-mobile/XCTest-Gherkin/commit/5aa806dd2163b481163c999accd2d6530990b566) and I can confirm it is not crashing for us in Xcode 8 anymore. 
Thanks a million!

**Comment by deanWombourne** [comment]
OK, I've released that commit as 0.10.5, and I'll close this issue. Thanks for your help!
